lib/rules/object-curly-spacing.js

```diff
@@ -121,9 +121,10 @@ module.exports = function(context) {
      * @returns {void}
      */
     function validateBraceSpacing(node, first, second, penultimate, last) {
+        var penultimateType = sourceCode.getNodeByRangeIndex(penultimate.start).type;
```

Comment:
Hmmm yeah, could be slow, especially because it will be called a bunch of times. Maybe we can speed up `getNodeByRangeIndex` by caching results?